### ISSUE TITLE
The detection results of the gitleaks rule using semgrep are different from that of the gitleaks  tool

### ISSUE THREAD

**Issue by KzDog** [returntocorp/sgrep-rules#2701]
**Describe the bug**
Taking the following string as an example, gitleaks will not be detected as a problem, but the gitleaks rule of semgrep will be detected as a problem

**To Reproduce**
1. Save the string "newPassword=this.mPassword" to the text.txt file
2. Use the command `semgrep scan --config semgrep-rules/semgrep/generic/secrets/gitleaks/generic-api-key.yaml` to scan and detect the problem
3. Use the command `gitleaks detect --no-git --source=. -v` to scan, will not be detected as a problem

**Expected behavior**
The test results of semgrep using the gitleaks rule are consistent with the gitleaks tool

**Comment by LewisArdern** [comment]
Hey @KzDog thanks for reporting it.

After further investigation this is because gitleaks does extra validation in their golang binary after detection:

Let's say gitleaks found `newPassword=this.mPassword` it would end up being `this.mPassword` and then they do some additional processing which is the following:

```
r, _ := regexp.Compile(...)
groups := r.FindStringSubmatch(...)
groups[1]

// look for stop words
password|passwd
// do entropy analysis
<false, so dont return>
```

Currently we have a lot of false positives with this rule mainly since we don't do the extra steps that gitleaks does. Since this is just a port of the regex.

I will add a fix for this shortly, https://semgrep.dev/s/RjBv this removes what you have above, I will look to also add the stop words, probably in a future addition since it does require some engine changes.

**Comment by LewisArdern** [comment]
it'll be fixed with next release of semgrep @KzDog
